.github/workflows/build_test_ci.yml

```diff
@@ -165,11 +166,30 @@ jobs:
       with:
         key: docker-${{ runner.os }}-${{ hashFiles('go.sum') }}
 
-    - name: E2E test
+    - name: Complete E2E Test
+      if: github.ref == 'refs/heads/main' 
       run: make e2etest
       env:
         GITHUB_TOKEN: ${{ secrets.github_token }}
         LINODE_TOKEN: ${{ secrets.LINODE_TOKEN }}
+        INSTALL_K3S_PROVIDER: true
```

Comment:
could we include these shared env variables at the job scope instead of the step scope?

Reply:
Hmmm GITHUB_TOKEN and LINODE_TOKEN can be moved to job level but other vars should be kept on step level because we would only be using them when complete e2e tests are triggered on main merge. 

What do you think?

Reply:
it looks like these are all duplicated between the two e2e tests
```
        LINODE_REGION: us-sea
        LINODE_CONTROL_PLANE_MACHINE_TYPE: g6-standard-2
        LINODE_MACHINE_TYPE: g6-standard-2
        CLUSTERCTL_CONFIG: /home/runner/work/cluster-api-provider-linode/cluster-api-provider-linode/e2e/clusterctl-config.yaml
```

Reply:
Oh that is because I wanted to run all the tests on this PR for testing purpose. I'm gonna remove those before we merge this PR to main. 

Only complete e2e test step should have those env vars

Reply:
ah, understood. ok you can ignore this then